script/bump_towncrier.py

```diff
@@ -0,0 +1,38 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
```

Comment:
Probably something that should be done in ``"python3 script/bump_changelog.py {new_version}"`` There's quite a bit of cleanup to do there with the adoption of towncrier ! We still have the script on the maintenance branch for 2.14 so almost everything can be removed (I think we only need to create the new file for minor automatically ?)

Reply:
I agree. I was a bit hesitant because of the "DO NOT MODIFY DIRECTLY" comment. 😄  That's probably not relevant any more / not relevant until we might decide to use `towncrier` in `android` as well?

Reading the `release.md` documentation, we have to handle two different invocations of `tbump`:

1. Invocation for a release version (without a `-devX`): 
1.1 Update the `version` in `towncrier.toml` 
1.2 build the final documentation 
2. Invocation for a dev version: 
2.1 Update the `version` in `towncrier.toml`
2.2 Update the `filename` in `towncrier.toml`
2.3 Create a new `doc/whatsnew/MAJOR/MAJOR.MINOR/index.rst` and link that in the parent `index.rst` files' toctrees

Maybe we should do the `tbump` related things in a separate PR, to keep this one reviewable.

Reply:
Hmm, right. Astroid is behind now and has a lot less activity than pylint. Last time I did astroid's script first, with the intention to use the same in both but pylint has different need. I don't mind keeping the old release and script routine in astroid as there is no whatsnew in astroid so no duplication for contributor.

Reply:
Did you have time to look at this @DudeNr33 ?

Reply:
Not yet unfortunately. I'll try to get to it in the next days.